File: 57-insert-interval/insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> result= new ArrayList<>();

        int n= intervals.length;
        int i=0;

        while(i<n && intervals[i][1]< newInterval[0]){
            result.add(intervals[i]);
            i++;
        }

        int mergeStart= newInterval[0];
        int mergeEnd= newInterval[1];

        while(i<n && intervals[i][0]<= mergeEnd){
            mergeStart= Math.min(mergeStart,intervals[i][0]);
            mergeEnd= Math.max(mergeEnd, intervals[i][1]);
            i++;
        }
        result.add(new int[]{mergeStart,mergeEnd});

        while(i<n ){
            result.add(intervals[i]);
            i++;
        }

        return result.toArray(new int[result.size()][]);
    }
}

File: 57-insert-interval/insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        n=len(intervals)
        result=[]
        i=0
        while i<n and intervals[i][1]<newInterval[0]:
            result.append(intervals[i])
            i+=1
        
        mergeStart= newInterval[0]
        mergeEnd= newInterval[1]

        while i<n and intervals[i][0]<=mergeEnd:
            mergeStart= min(mergeStart,intervals[i][0])
            mergeEnd= max(mergeEnd, intervals[i][1])
            i+=1
        result.append([mergeStart,mergeEnd])

        while(i<n):
            result.append(intervals[i])
            i+=1
        return result


        